libs/designer-ui/src/lib/floatingactionmenu/index.tsx

```diff
@@ -0,0 +1,148 @@
+    return (
+      <div role="button" className="msla-floating-action-menu" onClick={toggleExpanded} onKeyDown={toggleExpandedOnKeyDown} tabIndex={0}>
+        <span className="msla-floating-action-menu-plus-icon">{'+'}</span>
+        <span className="msla-floating-action-menu-title">{props.collapsedTitle}</span>
+      </div>
+    );
+  };
+
+  const renderMenuItems = (): JSX.Element => {
+    return (
+      <div className="msla-floating-action-menu-items-container">
+        {renderMenuItemsHeader()}
+        <div className="msla-floating-action-menu-items">
+          {props.menuItems.map((item: FloatingActionMenuItem) => {
+            return renderMenuItem(item);
+          })}
+        </div>
+      </div>
+    );
+  };
+
+  const renderMenuItemsHeader = (): JSX.Element => {
+    return (
+      <div>
+        <span className="msla-floating-action-menu-items-title">{props.expandedTitle}</span>
+        <span
+          role="button"
+          aria-label={closeErrorButtonAriaLabel}
+          className="msla-floating-action-menu-items-close"
+          onClick={toggleExpanded}
+          onKeyDown={toggleExpandedOnKeyDown}
+          tabIndex={0}
+        >
+          {'x'}
+        </span>
+      </div>
+    );
+  };
+
+  const renderMenuItem = (menuItem: FloatingActionMenuItem): JSX.Element => {
+    const itemStyle = {
+      background: `url('${menuItem.icon}') no-repeat center`,
+    };
+
+    if (props.menuItems.length === 1) {
+      return (
+        <div
+          role="button"
+          aria-label={menuItem.label}
+          tabIndex={0}
+          className="msla-floating-action-menu-item-vertical-container"
+          onClick={() => handleMenuItemSelected(menuItem)}
+          onKeyDown={(e: React.KeyboardEvent<HTMLElement>): void => handleMenuItemSelectedOnKeyDown(e, menuItem)}
+        >
+          <div className="msla-menu-item-logo" style={itemStyle} />
+          <span className="msla-vertical-menu-item-label">{menuItem.label}</span>
+          <div className="msla-vertical-menu-item-add" />
+        </div>
+      );
+    } else {
+      return (
+        <div key={menuItem.id} className="msla-floating-action-menu-item-horizontal-container">
+          <div
+            role="button"
+            aria-label={menuItem.label}
+            tabIndex={0}
+            className="msla-menu-item-logo"
+            style={itemStyle}
+            onClick={() => handleMenuItemSelected(menuItem)}
+            onKeyDown={(e: React.KeyboardEvent<HTMLElement>): void => handleMenuItemSelectedOnKeyDown(e, menuItem)}
+          />
+          <span className="msla-horizontal-menu-item-label">{menuItem.label}</span>
+        </div>
+      );
+    }
+  };
+
+  const handleMenuItemSelected = (item: FloatingActionMenuItem) => {
+    toggleExpanded();
+    props.onMenuItemSelected(item);
+  };
+
+  const handleMenuItemSelectedOnKeyDown = (e: React.KeyboardEvent<HTMLElement>, item: FloatingActionMenuItem) => {
+    const { keyCode } = e;
```

Comment:
VS Code told me `keyCode` is deprecated. Based on a quick search, we can use `e.code` instead. What is the  new recommended new way to identify keyboard events by keypress?

Reply:
I'm not sure, the go to seems to just be using string representations of the keys now. I don't particularly care for it either due to lack of type safety but should be fine as long as it's tested.